pkg/resources/create_resources.go

```diff
@@ -210,17 +211,21 @@ func (r *ResourceHelper) CreatePvcIfNotExists(owner v1.Object, pvc *corev1.Persi
 	return nil
 }
 
-func (r *ResourceHelper) DeleteIngress(key types.NamespacedName, reqLogger logr.Logger) error {
-	foundIngress := &networkingv1.Ingress{}
-	err := r.client.Get(context.TODO(), key, foundIngress)
-	if err != nil && k8sErrors.IsNotFound(err) {
+func (r *ResourceHelper) DeleteIngress(nameSpace string, labelSet map[string]string, reqLogger logr.Logger) error {
+	foundIngresses := &v1.PartialObjectMetadataList{}
+
+	err := r.client.List(context.TODO(), foundIngresses, client.InNamespace(nameSpace), &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labelSet),
+	})
+
+	if err != nil && len(foundIngresses.Items) > 0 {
 		return nil
 	} else if err != nil {
 		return errors.Wrap(err, "failed to check if ingress exists")
 	}
```

Comment:
Since only one result is expected, should we handle and raise an error if the list retrieves more than one result or should we remove all results? cc: @mattermost/cloud